processing/src/main/java/org/apache/carbondata/processing/loading/steps/DataWriterBatchProcessorStepImpl.java

```diff
@@ -141,7 +141,9 @@ private void finish(String tableName, CarbonFactHandler dataHandler) {
     try {
       processingComplete(dataHandler);
     } catch (Exception e) {
-      exception = new CarbonDataWriterException(e.getMessage(), e);
+      if (null == exception) {
```

Comment:
no need to keep the exception here. you can do the statistics in finally code block